### DISCUSSION TITLE
Get The (Actual) Current URL

### DISCUSSION THREAD

**Discussion by PoshoDev** [zauberzeug/nicegui#4000] [Q&A] [answered]
### Question

Hello there, 🙂

I'm trying to get the current URL that is being used on a browser for monitoring purposes on a site I'm working on. I'm currently using the following approach:

```python
from nicegui import context, ui


@ui.page("/{user}/{pet}")
def page_home(user: str, pet: str, source: str = None):
    print(context.client.page.path)

ui.run()
```
Browser URL:
```
http://127.0.0.1:8080/joe/truffles?source=local
```
Output:
```
/{user}/{pet}
```

The problem with this is that is that I can't truly see what the user accessed. I could implement some sort of parsing function that could interpret the parameter injection from the `ui.page` definition based on the variables' values, but there would still be the issue that if an user inserts a `?` parameter that I'm not contemplating I will never be able to see it. 🤔

Before I implement something, I want to ask if there's an actual proper `NiceGUI` approach for this.

Thanks for reading!

**Comment by falkoschindler** [comment, marked as answer]
Hi @PoshoDev,

Luckily you can access the current FastAPI [request](https://fastapi.tiangolo.com/reference/request/) object which contains the requested URL:
```py
@ui.page("/{user}/{pet}")
def page_home(user: str, pet: str, source: str = None):
    print(ui.context.client.request.url)
```
Apart from the complete URL it contains useful properties like `base_url`, `path_params`, `query_params` etc.

**Reply by PoshoDev** [reply]
Thank you very much! This is exactly what I needed! 😁
